conn2res/reservoir.py

```diff
@@ -31,23 +30,42 @@ class Conn:
     # TODO
     """
 
-    def __init__(self, subj_id=0, w=None):
+    def __init__(self, filename=None, subj_id=0, w=None):
         if w is not None:
             # assign provided connectivity data
             self.w = w
         else:
             # load connectivity data
-            self.w = load_file('connectivity.npy')
+            if filename is not None:
+                self.w = np.load(filename)
+            else:
+                self.w = load_file('connectivity.npy')
 
             # select one subject
             self.w = self.w[:, :, subj_id]
 
+        # set zero diagonal
+        np.fill_diagonal(self.w, 0)
+
+        # remove inf and nan
+        self.w[np.logical_or(np.isinf(self.w), np.isnan(self.w))] = 0
+
         # number of all active nodes
         self.n_nodes = len(self.w)
 
+        # number of edges (in symmetric networks edges are counted twice!)
+        self.n_edges = np.sum(self.w != 0)
+
+        # density of network
+        self.density = self.n_edges / (self.n_nodes * (self.n_nodes - 1))
+
         # indexes of set of active nodes
         self.idx_node = np.full(self.n_nodes, True)
 
+        # make sure that all nodes are connected to the rest of the network
```

Comment:
Maybe we could add something that checks for disconnected components too!

Reply:
To clarify, line 66 will eventually call self._get_largest_component, which checks all components in the network and if there are disconnected components then it takes the largest one and removes the others from the network. Is this what you had in mind @estefanysuarez or something different?